pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -275,7 +275,9 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) Results {
 	// We need to schedule them alternating, A, B, A, B, .... and this solution also solves that as well.
 	errors := map[*corev1.Pod]error{}
 	// Reset the metric for the controller, so we don't keep old ids around
-	UnschedulablePodsCount.DeletePartialMatch(map[string]string{ControllerLabel: injection.GetControllerName(ctx)})
+	if injection.GetControllerName(ctx) == "provisioner" {
```

Comment:
Can we just drop deleting this altogether? It's not clear to me why we just don't rely on the next round of scheduling to handle it -- the one thing that we do need to handle is setting the value to 0 when we don't find any pending pods/when we aren't getting a trigger to run scheduling.